src/core/core.tsx

```diff
@@ -21,271 +19,315 @@ import {
   OperationalError,
   NetworkError,
 } from '../errors';
-import { getTokens, TokenList } from '../tokens';
-
-// No need change the address, same is for all testnets and mainnet
-export const SWAP_ROUTER_ADDRESS = '0x68b3465833fb72A70ecDF485E0e4C7bD8665Fc45';
-export const ROUTER_API =
-  'https://a7sf9dqtif.execute-api.eu-central-1.amazonaws.com/prod';
-
-export const DEFAULTS = {
-  slippageTolerance: 1, // 1%
-  deadline: 200, // 200 seconds
-};
-
-export const SUPPORTED_CHAINS = [1, 4];
-
-const chainIdToNetwork: { [key: number]: Chain } = {
-  1: Mainnet,
-  4: Rinkeby,
-};
-
-const chainIDToNetworkInfo: { [key: number]: Info } = {
-  1: {
-    name: 'mainnet',
-    symbol: 'ETH',
-    decimals: 18,
-    wethAddress: '0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2',
-  },
-  4: {
-    name: 'rinkeby',
-    symbol: 'ETH',
-    decimals: 18,
-    wethAddress: '0xc778417E063141139Fce010982780140Aa0cD5Ab',
+import { getUniswapTokens, TokenInfo, TokenList } from '../tokens';
+import {
+  SWAP_ROUTER_ADDRESS,
+  ROUTER_API,
+  DEFAULTS,
+  SUPPORTED_CHAINS,
+  chainIdToNetwork,
+  chainIDToNetworkInfo,
+} from './constants';
+import { JsonRpcProvider } from '@ethersproject/providers';
+
+export const wallets = initializeWallets(SUPPORTED_CHAINS);
+
+export const config: Config = {
+  autoConnect: false,
+  notifications: {
+    expirationPeriod: 30000,
+    checkInterval: 2000,
   },
 };
 
-export class Layer2 {
-  public wallets: Wallet[];
-  public config: Config;
-  public interfaces: { [key: string]: Interface };
-  public defaults: { [key: string]: number };
+export const getQuote = async (
+  chainID: number,
+  inputAmount: number, // not formatted
+  tokenOut: string, // address
+  exactOut: boolean = false
+): Promise<QuoteDetails> => {
+  const tradeType = exactOut ? 'exactOut' : 'exactIn';
+  const formattedAmount = parseEther(inputAmount.toString()).toString();
 
-  constructor() {
-    this.wallets = initializeWallets(SUPPORTED_CHAINS);
-    this.defaults = DEFAULTS;
+  try {
+    const res = await fetch(
+      `${ROUTER_API}/quote?tokenInAddress=${chainIDToNetworkInfo[chainID].symbol}&tokenInChainId=${chainID}&tokenOutAddress=${tokenOut}&tokenOutChainId=${chainID}&amount=${formattedAmount}&type=${tradeType}`
+    );
+    const formattedResponse = await res?.json();
```

Comment:
Is there any scenario where `res` can be `undefined`? Maybe we can just `res?.json()` -> `res.json()`